src/Analysis/GWASResults/Views/Results/ResultsPheWeb/TopLociTable/TopLociTable.jsx

```diff
@@ -0,0 +1,187 @@
+import React, { useMemo, useState, useEffect } from 'react';
+import { Table, Input, Button } from 'antd';
+import PropTypes from 'prop-types';
+import { SearchOutlined } from '@ant-design/icons';
+import downloadTSVFromJson from './downloadTSVFromJSON';
+import filterLociTableData from './filterLociTableData';
+
+const TopLociTable = ({ data }) => {
+  // Adds a variant key value pair with the desired formatting
+  const tableData = useMemo(
+    () => data.map((obj) => ({
+      ...obj,
+      variant: `${obj?.chrom}:${obj?.pos.toLocaleString('en-US')} ${obj?.ref}/
+    ${obj?.alt} (${obj?.rsids})`,
+    })),
+    [data],
+  );
+
+  const [filteredData, setFilteredData] = useState(data);
+  const [lociTableState, setLociTableState] = useState({
+    variantSearchTerm: '',
+    nearestGenesSearchTerm: '',
+    pvalSearchTerm: '',
+    afSearchTerm: '',
+    currentPage: 1,
+  });
+
+  useEffect(() => {
+    setFilteredData(filterLociTableData(tableData, lociTableState));
+  }, [lociTableState, tableData]);
+
+  const handleSearchTermChange = (event, searchTermKey) => {
+    if (searchTermKey === 'variant') {
+      setLociTableState({
+        ...lociTableState,
+        currentPage: 1,
+        variantSearchTerm: event.target.value,
+      });
+    }
+    if (searchTermKey === 'nearest_genes') {
+      setLociTableState({
+        ...lociTableState,
+        currentPage: 1,
+        nearestGenesSearchTerm: event.target.value,
+      });
+    }
+    if (searchTermKey === 'pval') {
+      setLociTableState({
+        ...lociTableState,
+        currentPage: 1,
+        pvalSearchTerm: event.target.value,
+      });
+    }
+    if (searchTermKey === 'af') {
+      setLociTableState({
+        ...lociTableState,
+        currentPage: 1,
+        afSearchTerm: event.target.value,
+      });
+    }
+    return null;
+  };
+
+  const handleTableChange = (pagination) => {
+    if (pagination.current !== lociTableState.currentPage) {
+      // User changes page selection, set page to current pagination selection
+      return setLociTableState({
+        ...lociTableState,
+        currentPage: pagination.current,
+      });
+    }
+    // When the user updates sorting set page to first page
+    return setLociTableState({
+      ...lociTableState,
+      currentPage: 1,
+    });
+  };
+
+  const columns = [
+    {
+      title: 'Variant',
+      dataIndex: 'variant',
+      key: 'variant',
+      sorter: (a, b) => a.variant.localeCompare(b.variant),
+      children: [
+        {
+          title: (
+            <Input
+              placeholder='Search by Variant'
+              value={lociTableState.variantSearchTerm}
+              onChange={(event) => handleSearchTermChange(event, 'variant')}
```

Comment:
here we also might need a custom search. E.g. now when we have:
> 1:192,575,969 G/ A (rs7535818)

and search for `192575969` we get no hits. The value should match though, regardless of the locale formatting used in the field itself.

Reply:
Great idea. The search function has been omitted to strip commas from both the data at the key being searched and from the user input: 

```
const filterBySearchTerm = (data, key, searchTerm) => data.filter((obj) => obj[key]
  .toString()
  .toLowerCase()
  .replace(/,/g, '')
  .includes(searchTerm.toLowerCase().replace(/,/g, '')),
);
```